rust/src/fvm/machine.rs

```diff
@@ -307,3 +334,135 @@ fn import_actors(
     assert_eq!(roots.len(), 1);
     Ok(Some(roots[0]))
 }
+
+#[derive(Clone, Debug, Serialize_tuple, Deserialize_tuple)]
+struct LotusTrace {
+    pub msg: Message,
+    pub msg_receipt: Receipt,
+    pub error: String,
+    pub subcalls: Vec<LotusTrace>,
+}
+
+fn build_lotus_trace(
+    new_call: &ExecutionEvent,
+    trace_iter: &mut std::slice::Iter<ExecutionEvent>,
```

Comment:
```suggestion
    trace_iter: &mut impl Iterator<Item=ExecutionEvent>,
```

(makes your life easier).

Reply:
(assumes you use `into_iter`)